### DISCUSSION TITLE
Collabora Online

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#3651] [Announcements]
@remz1337 has brought us another great script, [Collabora Online](https://www.collaboraonline.com/).

Let's go through the normal testing to make sure it's ready for the website. Please post results.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/collabora-online.sh)"
```
It does need to be behind a proxy

**Comment by tteck** [comment]
Has anyone been able to get this going?

**Reply by knario** [reply]
Hi Tteck! 
I installed it and configured it with Nextcloud and it works very well, so far it has not given me any problem.

**Comment by Vakrehus** [comment]
```
   ______      ____      __                    
  / ____/___  / / /___ _/ /_  ____  _________ _
 / /   / __ \/ / / __ `/ __ \/ __ \/ ___/ __ `/
/ /___/ /_/ / / / /_/ / /_/ / /_/ / /  / /_/ / 
\____/\____/_/_/\__,_/_.___/\____/_/   \__,_/  
                                               
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: ********
Container ID: 173
Using Hostname: collabora
Using Disk Size: 12
Allocated Cores: 2
Allocated RAM: 1024
Using Bridge: vmbr0
Using IP Address: 192.168.1.173/24
Using Gateway IP Address: 192.168.1.1
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using Vlan: Default
Enable Root SSH Access: yes
Enable Verbose Mode: no
Creating a Collabora LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 173 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.1.173 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.121.3
 ✓ Updated Container OS
 | Installing Dependencies   
[ERROR] in line 23: exit code 0: while executing command "$@" > /dev/null 2>&1
The silent function has suppressed the error, run the script with verbose mode enabled, which will provide more detailed output.

root@pve:~# 

```

Will try again with verbose mode on.

**Comment by Vakrehus** [comment]
verbose mode ran through.

http://IP:9980 gives a simple "ok".

**Reply by tteck** [reply]
> http://ip:9980/ gives a simple "ok".

Here also

I used a Cloudflare tunnel, and got nowhere.

**Reply by remz1337** [reply]
yes the "ok" is expected, there's no UI for this. "Ok" means the websockets are working. On my side, once that worked, I installed the Nextcloud Office app on my Nextcloud server and configured it successfully (after configuring my reverse proxy to point to Collabora LXC)

**Reply by tteck** [reply]
That's not good. lol

**Reply by Vakrehus** [reply]
What does `http://IP:9980/browser` doing?

**Reply by remz1337** [reply]
from https://www.linode.com/docs/guides/how-to-install-collabora-code/

> 
> What is Collabora?
> ...
> Collabora can be installed on a private server or hosted in the cloud. It requires either a helper application like Nextcloud or a custom wrapper. Nextcloud is an open-source self-hosted file storage solution that emphasizes security and privacy. Nextcloud and Collabora have worked together to integrate the two applications. Within the Nextcloud application, Collabora CODE is known as Nextcloud Office.
> 
> To use Collabora from the Nextcloud interface, install both Collabora and Nextcloud and configure each application separately. Then, integrate Collabora into Nextcloud using the Nextcloud administrative hub. This allows users to view and edit Collabora documents directly within Nextcloud. When accessed in this manner, Collabora features full editing capabilities and enhanced security and is accessible from mobile devices.

**Reply by Vakrehus** [reply]
ahh, ok. it's just the code server. i thought it was a standalone office suite. I was hoping to throw MS Office overboard.

**Reply by tteck** [reply]
Unfortunately, since this script is different from the others, I believe it could cause confusion. With that in mind, I think I'll leave it off the website.

**Reply by remz1337** [reply]
Sounds good, thanks for the review
